src/diagram/components/quantity-node.tsx

```diff
@@ -1,96 +1,134 @@
-        {variable.operation}
-      </div>
-    </>
-    :
-    <>
-      Invalid Reference: {data.node.variableId}
-    </>;
+  const handleRemoveNode = () => {
+    const nodeToRemove = data.dqRoot.getNodeFromVariableId(variable.id);
+    data.dqRoot.removeNode(nodeToRemove);
+  };
+
+  const onValueChange = (evt: any) => {
+    // if the value is null or undefined just store undefined
+    if (evt.target.value == null) {
+      variable.setValue(undefined);
+    } else {
+      variable.setValue(parseFloat(evt.target.value));
+    }
+  };
+
+  const onUnitChange = (evt: any) => {
+    if (!evt.target.value) {
+      variable.setUnit(undefined);
+    } else {
+      variable.setUnit(evt.target.value);
+    }
+  };
 
+  const onNameChange = (evt: any) => {
+    if (!evt.target.value) {
+      variable.setName(undefined);
+    } else {
+      variable.setName(evt.target.value);
+    }
+  };
+  const onOperationChange = (evt: any) => {
+    if (!evt.target.value) {
+      variable.setOperation(undefined);
+    } else {
+      variable.setOperation(evt.target.value);
+    }
+  };
+  const shownValue = variable.numberOfInputs > 0 ? variable.computedValue : variable.value;
+  const shownUnit = variable.numberOfInputs > 0 ? variable.computedUnit : variable.unit;
+  const nodeHandleStyle = {border: "1px solid white", borderRadius: "6px", width: "12px", height: "12px", background: "#bcbcbc"};
   return (
-    <>
+    <div className={"node"} data-testid="quantity-node">
+      <div className="remove-node-button" onClick={handleRemoveNode} title={"Delete Node"} data-testid={"delete-node-button"}>
+        <DeleteIcon />
+      </div>
       <Handle
         type="target"
         position={Position.Left}
-        style={{ top: "27%", background: "#555" }}
+        style={{top: "30%", left: "-6px", ...nodeHandleStyle}}
         onConnect={(params) => console.log("handle onConnect", params)}
         isConnectable={isConnectable}
         id="a"
       />
       <Handle
         type="target"
         position={Position.Left}
-        style={{ top: "73%", background: "#555" }}
+        style={{ top: "70%", left: "-6px", ...nodeHandleStyle}}
         onConnect={(params) => console.log("handle onConnect", params)}
         isConnectable={isConnectable}
         id="b"
       />
-      <div style={{padding: "10px"}}>
-        {nodeInside}
+      <div className="variable-info-container">
+        <input className="variable-info name" type="text" placeholder="name" value={variable.name || ""}
+            onMouseDown={e => e.stopPropagation()} onChange={onNameChange} data-testid="variable-name"/>
+        <div className="variable-info-value-unit">
+          <input className="variable-info value" type="number" placeholder="value" onChange={onValueChange} data-testid="variable-value"
+            value={shownValue !== undefined ? shownValue.toString() : ""} onMouseDown={e => e.stopPropagation()} />
```

Comment:
This will make it so I can't demo the changes the unit code which support. Converting units better. However I can show this in a different branch. So I think this is OK to merge, and we'll deal with it later.